src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -1180,8 +1185,6 @@ private void RerouteAllQueuedMessages(ActivationData activation, ActivationAddre
                     throw new DuplicateActivationException(id);
                 }
             }
-            // We currently don't have any other case for multiple activations except for StatelessWorker.
```

Comment:
But no the following line with `//await scheduler.RunOrQueueTask`?

Or if you meant it to stay as an example of what we'd do in case of a future multi-activation case, I think the line is already incorrect because of the changed `RegisterAsync` behavior that now also includes what used to be `RegisterSingleActivationAsync`. Now it requires an additional bool argument.

I think we should remove that code line to avoid potential confusion but leave the comment line before it.